### PR TITLE
Course sorting 🎉 

### PR DESCRIPTION
**Pull request by bjrne**
@mathebox

### CODE REVIEW

File: xikolo-common/Data/Model/Course.swift [hunk 1]
```diff
@@ -83,6 +83,14 @@ class Course : BaseModel {
 
 }
 
+extension Course : DynamicSort {
+
+    func computeOrder() {
+        self.order = NSNumber(value: abs(start_at?.timeIntervalSinceNow ?? TimeInterval.infinity))
```
Comment (jprberlin) [inline xikolo-common/Data/Model/Course.swift]: what if start at is not set?

Comment (bjrne) [inline xikolo-common/Data/Model/Course.swift]: It's already in that line; it gets set to `infinity` and will end up at the bottom of the list.

**Comment by jprberlin** [comment]
Ok, so this is ready for merge i guess. @mathebox any objections?

File: xikolo-common/ViewHelpers/CollectionViewMultipleResultsControllerDelegateImplementation.swift [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+//  Copyright © 2016 HPI. All rights reserved.
+//
+
+import CoreData
+import UIKit
+
+class CollectionViewMultipleResultsControllerDelegateImplementation : NSObject, NSFetchedResultsControllerDelegate {
+
+    weak var collectionView: UICollectionView!
+    var resultsController: [NSFetchedResultsController<NSFetchRequestResult>] // 2Think: Do we create a memory loop here?
```
Comment (mathebox) [inline xikolo-common/ViewHelpers/CollectionViewMultipleResultsControllerDelegateImplementation.swift]: use: resultController**S**

File: xikolo-common/ViewHelpers/CollectionViewMultipleResultsControllerDelegateImplementation.swift [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            }
+        }, completion: nil)
+    }
+
+}
+
+extension CollectionViewMultipleResultsControllerDelegateImplementation : UICollectionViewDataSource {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return (resultsController.count == 1) ? 0 : resultsController.count
```
Comment (mathebox) [inline xikolo-common/ViewHelpers/CollectionViewMultipleResultsControllerDelegateImplementation.swift]: So we will not support multiple sections resultControllers?
If yes, it has to be documented.

**Comment by bjrne** [comment]
No bugs anymore (mostly thanks to @mathebox). One last round of reviews please @rhlff @jprberlin

**Comment by jprberlin** [comment]
This changes seems to break TVOS. So while this is good to merge please adapt tvOS as well in a different branch.

**Comment by mathebox** [comment]
Fixing tvOS should be easy.

**Pull request merged**